### PR TITLE
Enable full symbols for windows

### PR DESCRIPTION
**Pull request by TravisEz13**
## PR Summary

The main purpose of this was to enable full symbols for windows release build. 
I discovered an additional issue when trying to fix this:

- We were building Linux without optimization (debug) 

~~This PR addresses this my merging the Linux configuration into the release configuration and using a condition to add the build constant for unix~~
Update: optimizing on mac and Linux cause crashes.  An issue was filed to track this:  https://github.com/dotnet/corefx/issues/29700

Before we merge, I should validate that the release build works for:

- [x] Linux
- [x] mac
- [x] windows

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: PowerShell.Common.props [hunk 1]
```diff
@@ -108,4 +108,32 @@
...
+
+  <PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
+    <Optimize>true</Optimize>
+  </PropertyGroup>
+
+  <PropertyGroup Condition=" '$(Configuration)' == 'Release' And '$(IsWindows)' == 'true' ">
+    <DebugType>full</DebugType>
+  </PropertyGroup>
+
+  <PropertyGroup Condition=" '$(Configuration)' == 'Release' And '$(IsWindows)' == 'true' ">
```
Comment (adityapatwardhan) [inline PowerShell.Common.props]: true -> false

File: build.psm1 [hunk 2]
```diff
@@ -736,36 +736,20 @@ function New-PSOptions {
...
+                    $Configuration = "Debug"
                     Write-Warning ($ConfigWarningMsg -f $switch.Current, $Environment.LinuxInfo.PRETTY_NAME, $Configuration)
                 }
             }
         }
     }
     Write-Verbose "Using configuration '$Configuration'"
 
-    $PowerShellDir = if ($Configuration -eq 'Linux') {
+    $PowerShellDir = if ($Environment.IsLinux -or $Environment.IsMacOS) {
```
Comment (adityapatwardhan) [inline build.psm1]: can we do `-not $Environment.IsWindows` here as we elsewhere. Non-blocking.

File: src/Microsoft.PowerShell.Security/Microsoft.PowerShell.Security.csproj [hunk 3]
```diff
@@ -10,27 +10,11 @@
...
-
-  <PropertyGroup Condition=" '$(Configuration)' == 'Linux' ">
-    <DefineConstants>$(DefineConstants);UNIX</DefineConstants>
-  </PropertyGroup>
-
-  <PropertyGroup Condition=" '$(Configuration)' == 'CodeCoverage' ">
-    <DebugType>full</DebugType>
-  </PropertyGroup>
-
+
```
Comment (adityapatwardhan) [inline src/Microsoft.PowerShell.Security/Microsoft.PowerShell.Security.csproj]: Extra line?

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: PowerShell.Common.props [hunk 4]
```diff
@@ -113,26 +113,32 @@
...
     <Optimize>true</Optimize>
   </PropertyGroup>
 
+  <!-- Define windows, release configuration properties -->
   <PropertyGroup Condition=" '$(Configuration)' == 'Release' And '$(IsWindows)' == 'true' ">
     <DebugType>full</DebugType>
   </PropertyGroup>
 
-  <PropertyGroup Condition=" '$(Configuration)' == 'Release' And '$(IsWindows)' == 'true' ">
+  <!-- Define non-windows, release configuratio properties -->
```
Comment (adityapatwardhan) [inline PowerShell.Common.props]: typo configuratio -> configuration

**Review by adityapatwardhan — APPROVED** [review]

File: src/Microsoft.PowerShell.Security/Microsoft.PowerShell.Security.csproj (hunk 3 above)
Comment (TravisEz13) [reply src/Microsoft.PowerShell.Security/Microsoft.PowerShell.Security.csproj]: We usually have a blank line between groups.

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Comment by TravisEz13** [comment]
I filed an issue with dotnet for the issue encountered with optimizing non-windows platforms:  https://github.com/dotnet/corefx/issues/29700

File: PowerShell.Common.props [hunk 5]
```diff
@@ -108,4 +108,43 @@
...
+    <Optimize>true</Optimize>
+    <DebugType>full</DebugType>
+  </PropertyGroup>
+
+  <!-- Define non-windows, release configuration properties -->
+  <PropertyGroup Condition=" '$(Configuration)' == 'Release' And '$(IsWindows)' != 'true' ">
+    <!-- Set-Date fails with optimize enabled in NonWindowsSetDate
+         Debugging the issues resolves the problem
+     -->
+    <Optimize>false</Optimize>
```
Comment (iSazonov) [inline PowerShell.Common.props]: @TravisEz13 Should we set true because we fixed Set-Date?

**Pull request merged**